### DISCUSSION TITLE
Write ahead logging and pragma statements

### DISCUSSION THREAD

**Discussion by TheRocinante** [fnc12/sqlite_orm#775] [Q&A] [answered]
Hi!

I want to switch from the default journal mode and instead, I want to use write ahead logging.

I current define my schema and create the db using sqlite_orm. Is there a way to use the library to run a pragma statement, such as:

`PRAGMA journal_mode=WAL; `

If not, should I be manually creating the FB, running my pragma statements on it, and only then opening it using this library?

thanks so much.

**Comment by fnc12** [comment, marked as answer]
https://github.com/fnc12/sqlite_orm/blob/master/tests/pragma_tests.cpp#L26

**Reply by TheRocinante** [reply]
Oh excellent. I missed this file. I will try this out! Thank you.
